statescript/executor.go

```diff
@@ -205,6 +233,55 @@ func execute(name string, timeout time.Duration) error {
 	return nil
 }
 
+func retCode(err error) int {
+	defaultFailedCode := -1
+
+	if err != nil {
+		// try to get the exit code
+		if exitError, ok := err.(*exec.ExitError); ok {
+			ws := exitError.Sys().(syscall.WaitStatus)
+			return ws.ExitStatus()
+		} else {
+			return defaultFailedCode
+		}
+	}
+	return 0
+}
+
+// Catches a script that requests a retry in a loop. Is limited by the total window given to a script demanding a
+// retry.
+func executeScript(s os.FileInfo, dir string, l Launcher, timeout time.Duration, ignoreError bool) error {
+
+	iet := time.Now()
+	for {
+		err := execute(filepath.Join(dir, s.Name()), timeout)
+		switch ret := retCode(err); ret {
+		case 0:
+			// success
+			return nil
+		case exitRetryLater:
+			if time.Since(iet) <= l.getRetryTimeout() {
+				log.Infof("statescript: %s requested a retry", s.Name())
+				time.Sleep(l.getRetryInterval())
+				continue
+			}
+			if ignoreError {
+				log.Errorf("statescript: ignoring error executing '%s': %d: %s", s.Name(), ret, err.Error())
```

Comment:
@kacf this is related to the fixup you did for statescript-error messages. This should do the same thing, don't you think?

Reply:
Not sure which one you are referring to..?

Reply:
https://github.com/mendersoftware/mender/commit/f5ac3fcefced0c54ed348436f7e716bfa963e027

Reply:
Aah, that one. Yes, I believe so!